### DISCUSSION TITLE
What is the purpose of "samba samba-common-tools openssh sshpass" packages in docker image?

### DISCUSSION THREAD

**Discussion by thomaschampagne** [seriousm4x/UpSnap#515] [Q&A] [answered]
Hi,

Question is in the title :) 

Why do we need following packages: "samba samba-common-tools openssh sshpass" inside the production docker image ?

From a security aspect removing by hardening the image could reduce surface attack on this container.

Thanks for the help

**Comment by seriousm4x** [comment, marked as answer]
Samba packages are need for shutting down windows hosts via net rpc.

OpenSSH and sshpass are needed for shutting down linux hosts.

Both are essential for upsnap and don't offer much attach surface (normally - looking at you xz).

**Reply by thomaschampagne** [reply]
Make sense to me now. 

Thanks for your quick answer and your work :).
